lib/hostip.c

```diff
@@ -652,13 +652,31 @@ enum resolve_t Curl_resolv(struct Curl_easy *data,
   CURLcode result;
   enum resolve_t rc = CURLRESOLV_ERROR; /* default to failure */
   struct connectdata *conn = data->conn;
+  /* We should intentionally error and not resolve .onion TLDs */
+  bool found_onion = FALSE;
+  char *last_dot = strrchr(hostname, '.');
+  if(last_dot) {
+    char *stripped_hostname = strdup(hostname);
```

Comment:
Another solution to save an allocation would be to inspect `hostname` with `curl_strnequal()` and limiting the length in case of a trailing dot.

Reply:
We like avoiding allocations!